File: Main/ReferenceDLLs/Pack1/Facebook Connect Visual Kit CS/FVK Source Code/FVK/FVK/RequestDialog.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Text;

namespace FVK
{
    public partial class RequestDialog : System.Web.UI.UserControl
    {
        /// <summary>
        /// Set type of command control(link, button, auto_open). If auto_open is
        /// set permission popup is open on page load. Default = button.
        /// </summary>
        public string CommandType
        {
            set
            {
                switch (value)
                {
                    case "link": commandType = CommandTypeEnum.Link; break;
                    case "button": commandType = CommandTypeEnum.Button; break;
                    case "auto_open": commandType = CommandTypeEnum.AutoOpen; break;
                    default:
                        throw new Exception("Invalid command type, supported values are link, button and auto_open");
                }
            }
        }

        /// <summary>
        /// Set title of publish button/link
        /// </summary>
        public string CommandTitle { set { commandTitle = value; } }

        /// <summary>
        /// Set message of request dialog
        /// </summary>
        public string Message { set { message = value; } }

        /// <summary>
        /// Set title of request dialog
        /// </summary>
        public string Title { set { title = value; } }
        
        /// <summary>
        /// Set id of friend to which request has to be sent. If omited, friend selector is shown
        /// </summary>
        public string FriedId { set { friendId = value; } }

        /// <summary>
        /// Optional data which is sent with request
        /// </summary>
        public string AdditionalData { set { additionalData = value; }}

        /// <summary>
        /// Page Load
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        protected void Page_Load(object sender, EventArgs e)
        {
            Refresh();
        }

        private void Refresh()
        {
            string functionName = "ShowRequestDialog" + this.ClientID;
            StringBuilder html = new StringBuilder();
            html.Append("<script>\n");
            html.Append("function " + functionName + "() {\n");
            html.Append("  if (graphApiInitialized != true) {\n");
            html.Append("    setTimeout('" + functionName + "()', 100);\n");
            html.Append("    return;\n");
            html.Append("  }\n");
            html.Append("  FB.ui({method: 'apprequests', message: '");
            html.Append(message + "'");
            if (title != null)
            {
                html.Append(", title: '" + title + "'");
            }
            if (friendId != null)
            {
                html.Append(", to: '" + friendId + "'");
            }
            if (additionalData != null)
            {
                html.Append(", data: '" + additionalData + "'");
            }
            html.Append("});\n");
            html.Append("}\n");
            
            html.Append("</script>\n");

            switch (commandType)
            {
                case CommandTypeEnum.Link:
                    html.Append("<a style=\"cursor:pointer;\" onclick=\"" + functionName + "()\" >" );
                    html.Append(commandTitle);
                    html.Append("</a>\n");
                    break;
                case CommandTypeEnum.Button:
                    html.Append("<input type=\"button\" style=\"cursor:pointer\" class=\"button\" onclick=\"" + functionName + "()\"  value=\"");
                    html.Append(commandTitle);
                    html.Append("\" />\n");
                    break;
                case CommandTypeEnum.AutoOpen:
                    html.Append("<script type=\"text/javascript\">");
                    html.Append(functionName + "();\n");
                    html.Append("</script>\n");
                    break;
            }
            requestSpan.InnerHtml = html.ToString();
        }



        // Private members
        private string message;
        private string title;
        private string friendId;
        private string additionalData;
        private CommandTypeEnum commandType = CommandTypeEnum.Button;
        private string commandTitle = "Send Request";
          
    }
}

File: Main/ReferenceDLLs/Pack1/Facebook Visual Kit VB/FVK Source Code/FVK/FVK/RequestDialog.ascx.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Text

Namespace FVK
    Partial Public Class RequestDialog
        Inherits System.Web.UI.UserControl
        ''' <summary>
        ''' Set type of command control(link, button, auto_open). If auto_open is
        ''' set permission popup is open on page load. Default = button.
        ''' </summary>
        Public WriteOnly Property CommandType() As String
            Set(ByVal value As String)
                Select Case value
                    Case "link"
                        m_commandType = CommandTypeEnum.Link
                        Exit Select
                    Case "button"
                        m_commandType = CommandTypeEnum.Button
                        Exit Select
                    Case "auto_open"
                        m_commandType = CommandTypeEnum.AutoOpen
                        Exit Select
                    Case Else
                        Throw New Exception("Invalid command type, supported values are link, button and auto_open")
                End Select
            End Set
        End Property

        ''' <summary>
        ''' Set title of publish button/link
        ''' </summary>
        Public WriteOnly Property CommandTitle() As String
            Set(ByVal value As String)
                m_commandTitle = value
            End Set
        End Property

        ''' <summary>
        ''' Set message of request dialog
        ''' </summary>
        Public WriteOnly Property Message() As String
            Set(ByVal value As String)
                m_message = value
            End Set
        End Property

        ''' <summary>
        ''' Set title of request dialog
        ''' </summary>
        Public WriteOnly Property Title() As String
            Set(ByVal value As String)
                m_title = value
            End Set
        End Property

        ''' <summary>
        ''' Set id of friend to which request has to be sent. If omited, friend selector is shown
        ''' </summary>
        Public WriteOnly Property FriedId() As String
            Set(ByVal value As String)
                friendId = value
            End Set
        End Property

        ''' <summary>
        ''' Optional data which is sent with request
        ''' </summary>
        Public WriteOnly Property AdditionalData() As String
            Set(ByVal value As String)
                m_additionalData = value
            End Set
        End Property

        ''' <summary>
        ''' Page Load
        ''' </summary>
        ''' <param name="sender"></param>
        ''' <param name="e"></param>
        Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            Refresh()
        End Sub

        Private Sub Refresh()
            Dim functionName As String = "ShowRequestDialog" & Me.ClientID
            Dim html As New StringBuilder()
            html.Append("<script>" & vbLf)
            html.Append("function " & functionName & "() {" & vbLf)
            html.Append("  if (graphApiInitialized != true) {" & vbLf)
            html.Append("    setTimeout('" & functionName & "()', 100);" & vbLf)
            html.Append("    return;" & vbLf)
            html.Append("  }" & vbLf)
            html.Append("  FB.ui({method: 'apprequests', message: '")
            html.Append(m_message & "'")
            If m_title IsNot Nothing Then
                html.Append(", title: '" & m_title & "'")
            End If
            If friendId IsNot Nothing Then
                html.Append(", to: '" & friendId & "'")
            End If
            If m_additionalData IsNot Nothing Then
                html.Append(", data: '" & m_additionalData & "'")
            End If
            html.Append("});" & vbLf)
            html.Append("}" & vbLf)

            html.Append("</script>" & vbLf)

            Select Case m_commandType
                Case CommandTypeEnum.Link
                    html.Append("<a style=""cursor:pointer;"" onclick=""" & functionName & "()"" >")
                    html.Append(m_commandTitle)
                    html.Append("</a>" & vbLf)
                    Exit Select
                Case CommandTypeEnum.Button
                    html.Append("<input type=""button"" style=""cursor:pointer"" class=""button"" onclick=""" & functionName & "()""  value=""")
                    html.Append(m_commandTitle)
                    html.Append(""" />" & vbLf)
                    Exit Select
                Case CommandTypeEnum.AutoOpen
                    html.Append("<script type=""text/javascript"">")
                    html.Append(functionName & "();" & vbLf)
                    html.Append("</script>" & vbLf)
                    Exit Select
            End Select
            requestSpan.InnerHtml = html.ToString()
        End Sub



        ' Private members
        Private m_message As String
        Private m_title As String
        Private friendId As String
        Private m_additionalData As String
        Private m_commandType As CommandTypeEnum = CommandTypeEnum.Button
        Private m_commandTitle As String = "Send Request"

    End Class
End Namespace
